librz/include/rz_arch.h

```diff
@@ -59,6 +61,245 @@ RZ_API bool rz_arch_load_platform_sdb(RZ_NONNULL RzArchPlatformTarget *t, RZ_NON
 RZ_API bool rz_arch_platform_init(RzArchPlatformTarget *t, RZ_NONNULL const char *arch, RZ_NONNULL const char *cpu,
 	const char *platform, RZ_NONNULL const char *platforms_dir);
 
+typedef enum {
+	RZ_ARCH_INFO_CODE_ALIGN = 0, ///< code/text segment alignment
+	RZ_ARCH_INFO_DATA_ALIGN, ///< data segment alignment
+	RZ_ARCH_INFO_INSTRUCTION_MIN_SIZE, ///< minimum instruction length
+	RZ_ARCH_INFO_INSTRUCTION_MAX_SIZE, ///< maximum instruction length
+	RZ_ARCH_INFO_ADDRESS_SPACE, ///< address space bit-size
+	/* ignore */
+	RZ_ARCH_INFO_END, ///< Used only to define the max value of RZ_ARCH_INFO_* enum
+} RzArchInfo;
```

Comment:
Probably also endianess. Some architectures have multiple endianness variants, like PPC or ARM, sometimes even can switch on the fly, we should account for that.